examples/gno.land/p/demo/memeland/memeland.gno

```diff
@@ -0,0 +1,196 @@
+	UpvoteTracker *avl.Tree // address > bool hasVoted
+}
+
+type Memeland struct {
+	Posts       *avl.Tree
+	MemeCounter seqid.ID
+}
+
+func NewMemeland() *Memeland {
+	return &Memeland{
+		Posts: avl.NewTree(),
+	}
+}
+
+// PostMeme - Adds a new post
+func (m *Memeland) PostMeme(data, timestamp string) string {
+	if data == "" || timestamp == "" {
+		return "timestamp and data cannot be empty"
+	}
+
+	// Generate ID
+	id := m.MemeCounter.Next().String()
+
+	// Parse unix timestamp into int
+	t, err := strconv.Atoi(timestamp)
+	if err != nil {
+		return "cannot parse timestamp"
+	}
+
+	newPost := &Post{
+		ID:            id,
+		Data:          data,
+		Author:        std.GetOrigCaller(),
+		Timestamp:     time.Unix(int64(t), 0),
+		Upvotes:       0,
+		UpvoteTracker: avl.NewTree(),
+	}
+
+	m.Posts.Set(id, newPost)
+
+	return id
+}
+
+func (m *Memeland) Upvote(id string) string {
+	post, exists := m.Posts.Get(id)
+	if !exists {
+		return "post with specified ID does not exist"
+	}
+
+	p, ok := post.(*Post)
+	if !ok {
+		return "error casting post"
+	}
+
+	caller := std.GetOrigCaller().String()
+
+	if _, exists := p.UpvoteTracker.Get(caller); exists {
+		return "user has already upvoted this post"
+	}
+
+	p.Upvotes += 1
+	m.Posts.Set(id, p)
+	p.UpvoteTracker.Set(caller, true)
+
+	return "upvote successful"
+}
+
+// GetPost - Retrieves a post
+func (m *Memeland) GetPost(id string) (*Post, bool) {
+	value, exists := m.Posts.Get(id)
+	if exists {
+		return value.(*Post), true
+	}
+	return nil, false
+}
+
+func (m *Memeland) GetPosts() []*Post {
+	var posts []*Post
+
+	m.Posts.Iterate("", "", func(key string, value interface{}) bool {
+		post := value.(*Post)
+		posts = append(posts, post)
+
+		return false
+	})
+
+	return posts
+}
+
+// GetPostsByTimestamp returns a JSON string of posts within the given timestamp range, supporting pagination.
+func (m *Memeland) GetPostsByTimestamp(startTimestamp, endTimestamp string, page, pageSize int) string {
+	var filteredPosts []*Post
+
+	if page < 0 {
+		return "page count cannot be less than 1"
+	}
+
+	if pageSize < 0 {
+		return "page size cannot be less than 1"
+	}
+
+	startIndex := (page - 1) * pageSize
+	endIndex := startIndex + pageSize
+	count := 0
+
+	// Convert timestamps to time.Time
+	startTime, err := strconv.Atoi(startTimestamp)
+
+	if err != nil {
+		return "could not parse start timestamp"
+	}
+	endTime, err := strconv.Atoi(endTimestamp)
+	if err != nil {
+		return "could not parse end timestamp"
+	}
+
+	start := time.Unix(int64(startTime), 0)
+	end := time.Unix(int64(endTime), 0)
+
+	// Use AVL tree traversal to filter posts by timestamp range
+	m.Posts.Iterate("", "", func(key string, value interface{}) bool {
+		post := value.(*Post) // Type assertion, assuming value is always *Post
+		if post.Timestamp.After(start) && post.Timestamp.Before(end) {
+			if count >= startIndex && count < endIndex {
+				filteredPosts = append(filteredPosts, post)
+			}
+			count++
+		}
+		return false
```

Comment:
Please clean up this nesting

Reply:
Fixed it up: [61d2d12](https://github.com/gnolang/gno/pull/1751/commits/61d2d126f6fa12e09d8b8ff94ea6e0883a3188f4)